org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -241,8 +241,8 @@ private List<WorkspaceFolder> getRelevantWorkspaceFolders() {
 	}
 
 	/**
-	 * Starts a language server and triggers initialization. If language server is
-	 * started and active, does nothing. If language server is inactive, restart it.
+	 * Starts a language server and triggers initialization. If language server has been started
+	 * before, does nothing.
 	 */
 	public synchronized void start() {
 		start(false);
```

Comment:
If an exception has been occurred during start. The LS cannot be started again when the cause has been fixed (e.g. wrong path do LS binary).

Reply:
I think it could we worth explaining as well that "an Eclipse-base product can check `startupFailed` and call restart if the problem has been programmatically handled", and that otherwise "once the problem is fixed a restart of eclipse is needed". That would clarify as well the need for the `startupFailed` method for future readers of this class. My wording can most probably be improved :)

Reply:
..but I am okay with it. This can/should be handled by the LS provider